### ISSUE TITLE
[icalendar] eventfilter not returning event informations

### ISSUE THREAD

**Issue by rgrollfitz** [openhab/openhab-addons#9029]
The newly introduced version of the iCalendar binding for OH3 adds eventfilter. I tried this today with my Synology NAS calendar as the source but don't get any item state back for the events that should be grabbed via the eventfilter. Everything else is working as expected.

I used the binding config from the docs as you see [here](https://community.openhab.org/t/oh3-icalendar-eventfilter-not-bringing-up-tomorrows-events/108239)

I don't get any warnings or errors in the openhab.log. I'm using Build 2005.

**Comment by cweitkamp** [comment]
I am facing a similar problem but it only appears after a restart of openHAB. All my items are still `NULL`. It helps to disable and enable the Thing to get the values populated.

**Comment by rgrollfitz** [comment]
Thanks for the hint - but restart or disabling/enabling the thing/bridge does not work here. Even updated to build 2010 today - nothing changed.

**Comment by daMihe** [comment]
@rgrollfitz This may be a race condition (i can't reproduce directly). Are the properties set correctly after a "refreshTime" period?

**Comment by daMihe** [comment]
I've fixed a race condition which could lead to this behavior. That had to be fixed anyway. I got the behavior exactly once yesterday while debugging but i could not reproduce the behaviour getting always "NULL". If the issue still persists using icalendar including the fix (as your environment seems to favor this behaviour, should be included in next daily build thanks to @cpmeister), please feel free to reopen this issue. Please share then a minimal ical where the issue occurs.

**Comment by rgrollfitz** [comment]
Hey @daMihe 

sry for my late response. 
I looked into it today and I think I found the error. I'm using text-files only for configuration normaly but had a look in the UI today.

There were no linked channels in my thing configuration of the eventfilter visible. I used the example configuration from the docs (which was updated at that time, I think).
The example items config refered to 'event_0#xy' as the channel-scheme for the eventfilter - now I recognized that 'result_0#xy' would be the correct scheme here.

It's also mentioned in the decription itself correct now (I'm unsure it was there at the time I looked it up ) but the example item configuration still shows the wrong 'event_0#xy' scheme:

`String   first_event_name_tomorrow "first event [%s]"                         <calendar> { channel="icalendar:eventfilter:feedd0d0:event_0#title" }`

Thanks for your help and as least it's a good thing, that this pushes the race-condition fix a bit. :-)
